vaccine_feed_ingest/runners/il/sfsites/normalize.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+
+import datetime
+import json
+import pathlib
+import re
+import sys
+from typing import List, Optional
+
+import pydantic
+from vaccine_feed_ingest_schema import location as schema
+
+from vaccine_feed_ingest.utils.log import getLogger
+
+logger = getLogger(__file__)
+
+URL_RE = re.compile(
+    r"(https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]+\.[^\s]{2,}|www\.[a-zA-Z0-9]+\.[^\s]{2,})"
+)
+
+
+def _get_address(site: dict) -> schema.Address:
+    street1 = site["Street__c"]
+    city = site["City__c"]
+    zipc = site["Postal_Code__c"]
+
+    if not schema.ZIPCODE_RE.match(zipc):
+        zipc = None
+
+    return schema.Address(
+        street1=street1,
+        street2=None,
+        city=city,
+        state="IL",
+        zip=zipc,
+    )
+
+
+def _get_contact(site: dict) -> List[schema.Contact]:
+    contacts = []
+    url = site["Website__c"]
+    formatted_url = url
+
+    if " " in url:
+        match = URL_RE.search(url)
+
+        if match and match.group(1):
+            formatted_url = match.group(1)
+
+    if not formatted_url.startswith("http"):
+        formatted_url = "http://" + url
+
+    try:
+        contacts.append(
+            schema.Contact(
+                website=formatted_url, contact_type=schema.ContactType.BOOKING
+            )
+        )
+    except pydantic.ValidationError as e:
+        logger.warning(
+            "Invalid website for id: %s, value: %s, error: %s. Returning empty Contact",
+            site["Id"],
+            url,
+            str(e),
```

Comment:
Because pydantic errors are multi-line, this pattern can lead to noisy logs! It looks okay with this data source (the exception seems to be rarely hit), but please keep this in mind for future normalizers.

Reply:
I ended up removing the pydantic error - if someone wanted to debug, they can trivially get the error